modules/trino/src/main/java/org/testcontainers/containers/TrinoContainer.java

```diff
@@ -0,0 +1,109 @@
+package org.testcontainers.containers;
+
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.lang.String.format;
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+/**
+ * @author findepi
+ */
+public class TrinoContainer<SELF extends TrinoContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
+    public static final String NAME = "trino";
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("trinodb/trino");
+    public static final String IMAGE = "trinodb/trino";
+    public static final String DEFAULT_TAG = "351";
+
+    public static final Integer TRINO_PORT = 8080;
+
+    private String username = "test";
+    private String catalog = null;
+
+    public TrinoContainer(final String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    public TrinoContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+
+        this.waitStrategy = new LogMessageWaitStrategy()
+            .withRegEx(".*======== SERVER STARTED ========.*")
+            .withStartupTimeout(Duration.of(60, SECONDS));
+
+        addExposedPort(TRINO_PORT);
+    }
+
+    @NotNull
+    @Override
+    protected Set<Integer> getLivenessCheckPorts() {
```

Comment:
does not really need to be overridden?

Reply:
If a caller chooses to expose any additional ports (eg JMX), they should probably not become part of liveness probe automatically.